todoman/model.py

```diff
@@ -160,19 +160,40 @@ def clone(self):
     )
 
     def __setattr__(self, name, value):
-        # Avoids accidentally setting a field to None when that's not a valid
-        # attribute.
-        if not value:
-            if name in Todo.RRULE_FIELDS:
-                return object.__setattr__(self, name, '')
-            if name in Todo.STRING_FIELDS:
-                return object.__setattr__(self, name, '')
-            if name in Todo.INT_FIELDS:
-                return object.__setattr__(self, name, 0)
-            if name in Todo.LIST_FIELDS:
-                return object.__setattr__(self, name, [])
-
-        return object.__setattr__(self, name, value)
+        """Check type and avoid setting fields to None"""
+        """when that is not a valid attribue."""
+
+        v = value
+
+        if name in Todo.RRULE_FIELDS:
+            if value is None:
+                v = ''
+            elif not isinstance(value, str):
+                raise ValueError("Got {0} for {1} where str was expected"
```

Comment:
These validations are kinda unreachable, and are mostly here to prevent programming errors. Because of that, I'd much prefer to use `assert`, since otherwise these is normally-unreachable code.

Reply:
Hey, thanks for the review. :)

I externalized this part on purpose, because we will use it for the categories. (see PR #323)

Reply:
Done, changed it into assertions.